service/Dockerfile

```diff
@@ -0,0 +1,78 @@
+# LS docker file
+
+# build via (docker root = project root):
+# docker build -t absaoss/login-service:latest \
+# --build-arg BUILD_PROXY=http://my.cool.proxy.here:3128 \
+# --build-arg CONFIG=./path/to/my.awesome.local.application.yaml \
+# --build-arg SSL=true \
+# --build-arg SSL_DNAME="CN=*.my.domain.com, OU=project1, O=mycorp, L=Johannesburg, ST=Gauteng, C=za" .
+# run via
+# docker run -p 8080:8080 -p 8443:8443 absaoss/login-service:latest
+
+# test via:
+# http://localhost:8080/token/public-key
+# https://localhost:8443/token/public-key
+
+
+# --- Base image part (common for SSL true|false) ---
+# Conditional Docker image creation idea: https://stackoverflow.com/a/60820156/1773349
+# change to OFF to disable SSL
+ARG SSL=true
+
+# Tomcat OS base image, feel free to use another (e.g. your hardened one)
+ARG BASE_IMAGE=tomcat:9-jdk8-corretto
+FROM $BASE_IMAGE as base
+
+LABEL org.opencontainers.image.authors="ABSA"
+
+ARG BUILD_PROXY
+ARG CONFIG=./src/main/resources/example.application.yaml
+
+ENV LS_WAR_PATH=./target/scala-2.12
+# deploy as root
+ENV LS_WAR=ROOT.war
+
+# Copy Spring application properties
+COPY $CONFIG /opt/application.yaml
+
+COPY $LS_WAR_PATH/*.war /usr/local/tomcat/webapps/$LS_WAR
+
+ENV SPRING_CONFIG_ADDITIONAL-LOCATION=/opt/application.yaml
+
+ENV http_proxy=$BUILD_PROXY
+ENV https_proxy=$BUILD_PROXY
+ENV HTTP_PROXY=$BUILD_PROXY
+ENV HTTPS_PROXY=$BUILD_PROXY
+
+# uncomment and add packages you would need to install via yum if any
+#RUN yum -y update
+#RUN yum -y install htop procps
+
+# --- SSL=true image specifics ---
+FROM base AS base-ssl-true
+ENV SSL_ENABLED=true
+RUN echo "This stage sets SSL=$SSL_ENABLED"
+# Copy Tomcat properties (to enable SSL)
+COPY ./src/main/resources/tomcat-ssl/server.xml /usr/local/tomcat/conf/
+
+ARG SSL_DNAME="CN=*.my.example.com, OU=project1, O=yourcompany, L=Johannesburg, ST=Gauteng, C=za"
```

Comment:
I would just move all ARGs to the beginning of the Dockefile unless there is a good reason not to..?

Reply:
(but I will leave it up to you because I understand that you kind of want to put it close to the SSL related code)

Reply:
The thing here is that this arg is not used if SSL=false